src/components/v5/common/Fields/Input/Input.tsx

```diff
@@ -48,6 +49,21 @@ const Input: FC<InputProps> = ({
     }
   }, [isTyping, setIsTyping]);
 
+  const ref = useRef<HTMLInputElement | null>(null);
+
+  useLayoutEffect(() => {
+    if (ref.current) {
+      if (shouldFocus) {
+        ref.current.focus();
+      }
+
+      // Forward the ref to hook-form
+      registerField?.ref(ref.current);
+    }
+    // @NOTE: Including registerField would cause an infinite loop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [shouldFocus]);
```

Comment:
I'm using `useLayoutEffect` over regular effects after reading https://blog.octo.com/why-you-should-use-uselayouteffect-when-tinkering-with-focus

Reply:
Very nice